internal/api/mfa.go

```diff
@@ -196,6 +200,56 @@ func (a *API) ChallengeFactor(w http.ResponseWriter, r *http.Request) error {
 	})
 }
 
+func (a *API) invokeHook(ctx context.Context, input any, output any) *hooks.AuthHookError {
+	var response []byte
+	switch input.(type) {
+	case hooks.MFAVerificationAttemptInput:
+		payload, err := json.Marshal(&input)
+		if err != nil {
+			panic(err)
+		}
+		hookName, err := hooks.FetchHookName(a.config.Hook.MFAVerificationAttempt)
+		if err != nil {
+			return &hooks.AuthHookError{
+				Message: "invalid hook name",
+			}
+		}
+		if err := a.db.Transaction(func(tx *storage.Connection) error {
+			// We rely on Postgres timeouts to ensure the function doesn't overrun
+			timeoutQuery := tx.RawQuery(fmt.Sprintf("set local statement_timeout TO '%d';", hooks.DefaultTimeout))
+			if terr := timeoutQuery.Exec(); terr != nil {
+				return terr
+			}
+			query := tx.RawQuery(fmt.Sprintf("SELECT * from %s(?)", hookName), payload)
```

Comment:
Just wondering if these can be connected in one single query in a [DO]() block?

Also can you check with PostgreSQL team whether this statement_timeout can be somehow reset to the original value. Remember these connections are reused by GoTrue's DB pool so I don't want this timeout to leak across other users of the connection.

Reply:
Actually since it's a local statement timeout it should automatically reset at the end of the transaction